1_BC_StakeHolderRequirements

olafhuerta97 opened the pull request:
# 1_BC_StakeHolderRequirements
<!--- For the title and name of the PR use the following structure: #issue_TypeIssue_BriefDescription

TypeIssue is BC for Bug Fix or change, NF for New Feature or TE for Testing -->

## Description
<!---Please describe the problem, root cause and the solution provided. 
Include a summary of the change(s) and which issue is fixed or closed (optional).-->
Add Stakeholder Requirements

Fixes #1 


## Checklist:

- [x] Select asignee(s)
- [x] Select labels (optional)
- [x] Associate to the project
- [x] Select reviewer(s)
- [x] Link the issue
- [x] Ask for review
- [x] Make changes if needed
- [x] Merge the PR
- [x] Close the PR

olafhuerta97 committed:
Create StakeHolderRequirements.md

Added first draft

olafhuerta97 committed:
Added first draft of stakeholder requirements

olafhuerta97 commented:
@danycmtz @Hector19971997 @JDRQ98  please check this stakeholder requirements and let me know if you have comments or issues. This are not yet final but I think could be enough to trigger system requirements develop. Please think of another possible stakeholder requirement. 
> We are our own clients for now

JDRQ98 committed:
Organized requirements by categories

olafhuerta97 commented:
@danycmtz please add HW stakeholder requirements 
> as client

JDRQ98 committed:
Fixed typos

JDRQ98 commented:
Seems good to me.

danycmtz commented:
> @danycmtz please add HW stakeholder requirements
> 
> > as client

I think it's not necessary, every HW requirements are included for now.

danycmtz commented:
@JDRQ98 Thanks for the organization!

danycmtz reviewed (dismissed):
Seems great!

Hector19971997 committed:
Stakeholders requierements POV

- Added ID´s for each requirement for better traceability
- Added AT-9  to actuators.
- Created a review document
- Added findings in the review document

olafhuerta97 commented:
@danycmtz  could yo please help us by adding the template for this PR and the #11 too manually please

olafhuerta97 committed:
Merge pull request #11 from olafhuerta97/Update-SH-Requirements

1-Stakeholders requierements POV

olafhuerta97 commented:
@JDRQ98 could you please solve this PR and ask for review to @Hector19971997 so he can merge this?

JDRQ98 committed:
Revision of StakeHolderRequirements.md as per review document

danycmtz commented on Documentation/StakeHolderRequirements.md:
```diff
@@ -1,34 +1,34 @@
 ## Sensors:
-#### SNS-1 The truck must take pictures or videos and store in a SD memory
-#### SNS-2 The truck must record audio and save it
+#### SNS-1 The truck must be able to take pictures or videos and store it in an SD memory
```
I know it's a small detail, but I think this should be "The truck must be able to take pictures or videos and store them in an SD memory."

Everything else excellent job!

olafhuerta97 commented:
@Hector19971997 Please check latest changes and add comments if needed.

Hector19971997 commented:
@olafhuerta97 @JDRQ98 The review sheet was added to see the findings, but it also serves the purpouse of responding the open points there with a better traceability and add a reason to the change.
Also, I deleted the "be able to" in the "must be able to" since it still sound like an option, requirements shall be more firm(?), it sound the same as "may", it is not optional, its mandatory.

Hector19971997 commented:
@JDRQ98 Also, I saw the following:
#### COM-3 The truck state must be communicated via can bus through bluetooth
For COM-3, I was expecting instead going for something simpler like this:
COM-3 The truck modules must be communicated thought can bus /*The bluetooth diagnostic is now mentioned in COM-2, there being multiple requirement for the same thing is not the best, also there was no requirement for the communication between modules,*/

JDRQ98 committed:
Responding to findings in Review_StakeholderRequirements.md and updating StakeholderRequirements.md

JDRQ98 commented:
@Hector19971997 Made some updates as per your last comments, please have a look and let me know if anything else needs changing

olafhuerta97 commented on Documentation/StakeHolderRequirements.md:
```diff
@@ -0,0 +1,37 @@
+## Actuators:
+#### AT-1 The truck must have the ability to drive forward and in reverse
+#### AT-2 The truck must have the ability to change direction while going forward, reverse or, being idle.  
+#### AT-3 The truck must have right and left turn lights
```
This turn light should be trigger automaticlly by the steer, or manually, configurable?

JDRQ98 replied:
We could add both 'Auto' and 'Manual' mode for the turn lights

olafhuerta97 commented on Documentation/StakeHolderRequirements.md:
```diff
@@ -0,0 +1,37 @@
+#### AT-1 The truck must have the ability to drive forward and in reverse
+#### AT-2 The truck must have the ability to change direction while going forward, reverse or, being idle.  
+#### AT-3 The truck must have right and left turn lights
+#### AT-9 The truck must have headlights
```
Should this be trigger manually or by a light sensor?

JDRQ98 replied:
I think there should be three modes - 'on' 'off' and 'auto' which uses a light sensor. Making appropriate changes.

olafhuerta97 commented on Documentation/StakeHolderRequirements.md:
```diff
@@ -0,0 +1,37 @@
+#### AT-9 The truck must have headlights
+#### AT-4 The truck must have lights on the roof
+#### AT-5 The truck must have the ability to load and unload the semi Trailer
+#### AT-6 The truck must have the ability to play audio signals
```
from where?

JDRQ98 replied:
Nonvolatile memory

olafhuerta97 commented on Documentation/StakeHolderRequirements.md:
```diff
@@ -0,0 +1,37 @@
+#### AT-4 The truck must have lights on the roof
+#### AT-5 The truck must have the ability to load and unload the semi Trailer
+#### AT-6 The truck must have the ability to play audio signals
+#### AT-7 The truck must stop automatically if an obstacle is detected
```
going forward, backwards? both? This will be configurable?

JDRQ98 replied:
Automatic stop would be triggered by a LiDAR sensor and should be able to be turned on or off.  I think automatic stop should be only for forward driving.

Perhaps for reverse driving we could add another LiDAR to be used as a proximity sensor that lets you know how close you are to an obstacle through different frequency of beeps? Thoughts? @olafhuerta97 @danycmtz @Hector19971997

Hector19971997 replied:
I agree with @JDRQ98 Only automatic stop should be used for going foward and a "beep" sound while going backwards

olafhuerta97 replied:
Frequency of beeps will be then. And only going forwards automatic stop will be an option. Please update this so we can close this conversation.

danycmtz replied:
I agree the best option for going backwards is a frequency of beeps, not automatic stop, just like a real truck. 😃

olafhuerta97 commented on Documentation/StakeHolderRequirements.md:
```diff
@@ -0,0 +1,37 @@
+#### AT-5 The truck must have the ability to load and unload the semi Trailer
+#### AT-6 The truck must have the ability to play audio signals
+#### AT-7 The truck must stop automatically if an obstacle is detected
+#### AT-8 The truck must stop automatically if the weight of the load exceeds 1 pound
```
Is this configurable too?

olafhuerta97 commented on Documentation/StakeHolderRequirements.md:
```diff
@@ -0,0 +1,37 @@
+# Stakeholder requirements
+
+## Actuators:
+#### AT-1 The truck must have the ability to drive forward and in reverse
```
Is only forward or will it hable a "gas pedal" (speed control)

JDRQ98 replied:
Adding 'AT-10: The truck must have an analogic control system for velocity' after AT-1

Hector19971997 commented on Documentation/StakeHolderRequirements.md:
```diff
@@ -0,0 +1,35 @@
+LCD screen 20x4
+Radiofrequency module
+SD memory module
+Bocine
```
Bocine? Dont you mean speaker?

JDRQ98 committed:
Added requirements AT-9, AT-10, SNS-10 and SNS-11. Potentially final revision.

olafhuerta97 commented:
@Hector19971997 please check and if no more comments merge this PR

Hector19971997 commented on Documentation/StakeHolderRequirements.md:
```diff
@@ -2,35 +2,40 @@
+#### AT-4 The truck must have lights on the roof. 'On', 'Off' and 'Auto' (based on light sensor) configurations
+#### AT-5 The truck must have the ability to load and unload the semi trailer
+#### AT-6 The truck must have the ability to play audio signals stored in nonvolatile memory
+#### AT-9 While reversing, the truck must inform the user of proximity to a rear obstacle using a beeping sound. 'On' and 'Off' configurations
```
AT-9 is already in use, please change to AT-11

Hector19971997 committed:
Review small fix

Fixed ID for the review
